src/routes/(app)/person/[id]/+page.svelte

```diff
@@ -31,33 +34,64 @@
     return unsubscribe;
   });
 
-  $: {
-    (async () => {
-      try {
-        person = undefined;
-        pageError = undefined;
-        if (!personId) {
-          return;
-        }
-        const data = await getPerson(personId);
-        person = data;
-      } catch (err: any) {
-        person = undefined;
-        pageError = err;
+  $: if (personId) {
+    fetchPersonData();
+  }
+
+  $: if (sortOption && cachedCredits) {
+    sortedCredits = getSortedCredits(cachedCredits, sortOption);
+  }
+
+  async function fetchPersonData() {
+    try {
+      person = undefined;
+      pageError = undefined;
+      if (!personId) {
+        return;
       }
-    })();
+      person = await getPerson(personId);
+      cachedCredits = await getPersonCredits();
+      sortedCredits = getSortedCredits(cachedCredits, sortOption);
+    } catch (err: any) {
+      person = undefined;
+      pageError = err;
+    }
   }
 
   async function getPerson(id: number) {
     return (await axios.get(`/content/person/${id}`)).data as TMDBPersonDetails;
   }
 
   async function getPersonCredits() {
-    const credits = (await axios.get(`/content/person/${data.personId}/credits`))
+    let credits = (await axios.get(`/content/person/${data.personId}/credits`))
       .data as TMDBPersonCombinedCredits;
-    credits.cast = credits.cast?.sort((a, b) => b.vote_count - a.vote_count);
     return credits;
   }
+
+  function getSortedCredits(credits: TMDBPersonCombinedCredits, sortOption: string) {
+    if (!credits || !credits.cast) return;
+    let sorted = [...credits.cast];
+    switch (sortOption) {
+      case "Vote count":
+        sorted.sort((a, b) => b.vote_count - a.vote_count);
+        break;
+      case "Newest":
+        sorted.sort(
+          (a, b) =>
+            new Date(b.release_date || b.first_air_date).valueOf() -
+            new Date(a.release_date || a.first_air_date).valueOf()
+        );
+        break;
+      case "Oldest":
+        sorted.sort(
+          (a, b) =>
+            new Date(a.release_date || a.first_air_date).valueOf() -
+            new Date(b.release_date || b.first_air_date).valueOf()
+        );
+        break;
+    }
+    return { ...credits, cast: sorted };
```

Comment:
with the one array, i believe we shouldn't have to return anything - though possibly we may need to reassign credits = credits to get it to update ui (not 100% certain on that)